test/integration/http_integration.h

```diff
@@ -101,9 +101,12 @@ class HttpIntegrationTest : public BaseIntegrationTest {
       const Http::TestHeaderMapImpl& request_headers, uint32_t request_body_size,
       const Http::TestHeaderMapImpl& response_headers, uint32_t response_body_size);
 
-  // Wait for the end of stream on the next upstream stream on fake_upstreams_
+  // Wait for the end of stream on the next upstream stream on either of the two fake_upstreams_
   // Sets fake_upstream_connection_ to the connection and upstream_request_ to stream.
-  void waitForNextUpstreamRequest(uint64_t upstream_index = 0);
+  // In cases where the upstream that will receive the request is not deterministic, a second
+  // upstream index may be provided, in which case both upstreams will be checked for requests.
+  uint64_t waitForNextUpstreamRequest(uint64_t upstream_index = 0,
```

Comment:
If feels a bit clunky making this either 1 or 2; why not just provide a set of hosts?